client/src/components/DropdownMenu.js

```diff
@@ -24,46 +49,46 @@ export class DropdownMenu extends React.Component {
       dropdown_content_class_name,
       button_description,
       dropdown_trigger_txt,
+      dropdown_a11y_txt, //used if the trigger text is not a string object
     } = this.props;
     const { isOpen } = this.state;
 
+    const aria_label = _.isObject(dropdown_trigger_txt)
+      ? dropdown_a11y_txt
+      : dropdown_trigger_txt;
+
     return (
-      <div className="dropdown">
+      <div className="dropdown" ref={this.dropdown_ref}>
         <button
```

Comment:
The button should have `aria-haspopup="true"` and `aria-expanded={is_open}`

Reply:
With `aria-expanded` set you shouldn't need to manually report open/closed in the labels.